packages/devtools-reps/src/reps/tests/promise.js

```diff
@@ -0,0 +1,208 @@
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "pending" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with string", () => {
+  const object = stubs.get("FulfilledWithString");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: "foo" }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with a string", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with object", () => {
+  const object = stubs.get("FulfilledWithObject");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: Object }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with an object", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test Promise fulfilled with array", () => {
+  const object = stubs.get("FulfilledWithArray");
+  const defaultOutput = `Promise { <state>: "fulfilled", <value>: [3] }`;
+
+  it("correctly selects PromiseRep Rep for Promise fulfilled with an array", () => {
+    expect(getRep(object)).toBe(PromiseRep.rep);
+  });
+
+  it("should render as expected", () => {
+    expect(renderRep(object, {mode: undefined}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.TINY}).text())
+      .toBe('Promise { "fulfilled" }');
+    expect(renderRep(object, {mode: MODE.SHORT}).text())
+      .toBe(defaultOutput);
+    expect(renderRep(object, {mode: MODE.LONG}).text())
+      .toBe(defaultOutput);
+  });
+});
+
+describe("test on DOM node MouseOver", () => {
+  const stub = stubs.get("FulfilledWithNode");
+  const grips = getSelectableInInspectorGrips(stub);
+
+  it("has one node grip", () => {
+    expect(grips.length).toBe(1);
+  });
+
+  it("calls with expected argument when mouseover is fired on the node element", () => {
+    let mouseOverValue;
+    const onDOMNodeMouseOver = (object) => {
```

Comment:
Would you mind converting this to the "jest" way, which would look something like: 
```javascript
it("calls the expected function on mouseover", () => {
  const onDOMNodeMouseOver = jest.fn();
  const wrapper = renderRep(stub, {onDOMNodeMouseOver});

  wrapper.simulate("mouseover");

  expect(onDOMNodeMouseOver.mock.calls.length).toBe(1);
  expect(onDOMNodeMouseOver).toBeCalledWith(object);
});
```

`jest.fn()` gives us a way to control more deeply how and how many times the function was called (here's the doc for it https://facebook.github.io/jest/docs/mock-functions.html#content)